internal/sbom/convert/file/file.go

```diff
@@ -57,6 +57,13 @@ func ToComponent(logger zerolog.Logger, absFilePath, relFilePath string, options
 	}
 	component.Version = fmt.Sprintf("v0.0.0-%s", hashes[0].Value[:12])
 
+	if pathEnabled {
+		if component.Properties == nil {
+			component.Properties = &[]cdx.Property{}
+		}
+		*component.Properties = append(*component.Properties, sbom.NewProperty("path", absFilePath))
```

Comment:
Yeah, I actually wanted to add relative path first. However, I couldn't decide which strip path prefix should be. Because there's git repo with main code, external packages in GOROOT, some /usr/... paths... So, I decided that stripping file paths is often a specific user case. I was thinking to strip them to CI root after BOM generation.

Reply:
Maybe, we can optionally accept a base path in -paths BASE_PATH argument?

Reply:
The respective module's path is accessible via `gomod.Module`:

https://github.com/CycloneDX/cyclonedx-gomod/blob/d614aefe036344765fd7c76b1d9d9c08fceee3f5/internal/gomod/module.go#L45

A `Module` instance is passed to `github.com/CycloneDX/cyclonedx-gomod/internal/sbom/convert/pkg.Option`:

https://github.com/CycloneDX/cyclonedx-gomod/blob/d614aefe036344765fd7c76b1d9d9c08fceee3f5/internal/sbom/convert/pkg/package.go#L31

So the `github.com/CycloneDX/cyclonedx-gomod/internal/sbom/convert/pkg.WithFiles` option has access to it:

https://github.com/CycloneDX/cyclonedx-gomod/blob/d614aefe036344765fd7c76b1d9d9c08fceee3f5/internal/sbom/convert/pkg/package.go#L33-L37

It's simply not passed to `github.com/CycloneDX/cyclonedx-gomod/internal/sbom/convert/file.Option` yet:

https://github.com/CycloneDX/cyclonedx-gomod/blob/d614aefe036344765fd7c76b1d9d9c08fceee3f5/internal/sbom/convert/file/file.go#L29

But because this is entitely `internal` code, you could just add it to `Option` without breaking any API contracts.

Reply:
Thx! I'll rework the code